Transformer/UI/macOS/main.swift

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AppKit
+
+import Dispatch
+import Foundation
+
+internal class NSLabel: NSTextField {
+  public override init(frame: NSRect) {
+    super.init(frame: frame)
+
+    self.drawsBackground = false
+    self.isBezeled = false
+    self.isEditable = false
+    self.isSelectable = false
+  }
+
+  public required init?(coder: NSCoder) {
+    fatalError("init?(coder:) not implemented")
+  }
+}
+
+internal extension NSLabel {
+  convenience init(frame: NSRect, title: String) {
+    self.init(frame: frame)
+    self.stringValue = title
+  }
+}
+
+internal extension NSButton {
+  convenience init(frame: NSRect, title: String) {
+    self.init(frame: frame)
+    self.title = title
+  }
+
+  var text: String {
+    get { return self.stringValue }
+    set { self.stringValue = newValue }
+  }
+}
+
+internal extension NSTextField {
+  var text: String {
+    get { return self.stringValue }
+    set { self.stringValue = newValue }
+  }
+}
+
+internal extension NSTextView {
+  var text: String {
+    get { return self.string }
+    set { self.string = newValue }
+  }
+}
+
+internal extension NSSlider {
+  var value: Double {
+    get { return self.doubleValue }
+    set { self.doubleValue = newValue }
+  }
+}
+
+func onMain(_ body: @escaping () -> ()) {
+  if NSThread.isMainThread {
```

Comment:
```suggestion
  if Thread.isMainThread {
```
NSThread was deprecated in Swift 3, and this throws errors in an Xcode build.